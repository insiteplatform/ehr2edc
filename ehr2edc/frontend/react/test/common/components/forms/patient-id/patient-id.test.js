import PatientIdField from "../../../../../src/common/components/forms/patient-id/patient-id";
import React from "react";

describe("Rendering the PatientId input", () => {
    test("Renders without availablePatientIds", () => {
        const field = shallow(<PatientIdField/>);

        expect(field).toMatchSnapshot();
    });

    test("Renders with availablePatientIds", () => {
        const field = shallow(<PatientIdField availablePatients={{
            patientIds: ["PATIENT-1", "PATIENT-2"]
        }}/>);

        expect(field).toMatchSnapshot();
    });

    test("Renders with errors", () => {
        const field = shallow(<PatientIdField valid={false} availablePatients={{
            patientIds: ["PATIENT-1", "PATIENT-2"]
        }}/>);

        expect(field).toMatchSnapshot();
    })
});

describe("Interaction with parent component", () => {
    test("Callback is triggered onChange", () => {
        const callback = jest.fn();
        const field = shallow(<PatientIdField onChange={callback}/>);

        field.instance().patientIdChange("changedId");

        expect(callback).toHaveBeenCalledWith("changedId");
    });

    test('callback is triggered with value of the input field when the value is in the auto complete list', () => {
        const callback = jest.fn();
        const patientIdField = mount(<PatientIdField onChange={callback} availablePatients={{
            patientIds: ["PATIENT-1231", "PATIENT-1232", "PATIENT-1233"]
        }}/>);

        const input = patientIdField.find('input');
        input.simulate('change', {target: {value: 'PATIENT-1233'}})

        expect(callback).toHaveBeenCalledWith("PATIENT-1233");
    });

    test('callback is triggered with value of the input field when the value is NOT in the auto complete list', () => {
        const callback = jest.fn();
        const patientIdField = mount(<PatientIdField onChange={callback} availablePatients={{
            patientIds: ["PATIENT-1231", "PATIENT-1232", "PATIENT-1233"]
        }}/>);

        const input = patientIdField.find('input');
        input.simulate('change', {target: {value: 'PATIENT-1235'}});

        expect(callback).toHaveBeenCalledWith("PATIENT-1235");
    });
});
