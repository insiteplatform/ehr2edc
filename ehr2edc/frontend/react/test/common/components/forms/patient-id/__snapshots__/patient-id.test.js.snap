// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the PatientId input Renders with availablePatientIds 1`] = `
<AutocompleteInput
  alwaysRender={true}
  inputProps={
    Object {
      "className": "patient-identifier-field",
    }
  }
  onSelectionChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "label": "PATIENT-1",
        "value": "PATIENT-1",
      },
      Object {
        "label": "PATIENT-2",
        "value": "PATIENT-2",
      },
    ]
  }
  placeholder="Select EHR ID"
/>
`;

exports[`Rendering the PatientId input Renders with errors 1`] = `
<AutocompleteInput
  alwaysRender={true}
  inputProps={
    Object {
      "className": "patient-identifier-field form-field-error",
    }
  }
  onSelectionChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "label": "PATIENT-1",
        "value": "PATIENT-1",
      },
      Object {
        "label": "PATIENT-2",
        "value": "PATIENT-2",
      },
    ]
  }
  placeholder="Select EHR ID"
/>
`;

exports[`Rendering the PatientId input Renders without availablePatientIds 1`] = `
<AutocompleteInput
  alwaysRender={true}
  inputProps={
    Object {
      "className": "patient-identifier-field",
    }
  }
  onSelectionChange={[Function]}
  openOnFocus={true}
  options={Array []}
  placeholder="Select EHR ID"
/>
`;
